src/Integration.Vsix.UnitTests/SonarLintTagger/CancellableJobRunnerTests.cs

```diff
@@ -141,6 +141,8 @@ public void CancelAfterFirstOperation()
 
 
             WaitForRunnerToFinish(testSubject, testLogger);
+            // Pause for any final progress steps to be reported before checking the progressRecorder below
+            Thread.Sleep(200);
```

Comment:
@duncanp-sonar I'm not arguing with the fix, but can you explain why is this happening?
I see in the code that CancellableJobRunner.RunnerState is being changed before the token is cancelled, and the test progressRecorder is a sync operation. So what is it that we need to wait for?

Reply:
The test thread can restart running as soon as the `Cancel` operation is performed, so there are now two threads running:
a) the `CancellableJobRunner` thread that calls the `progressRecorder` to write the final "job cancelled" message, and
b) the test thread that is checking for that "job cancelled" message.

The test fails intermittently if (b) happens before (a). The delay is just to reduce/remove the number of times this happens.